lib/nationbuilder/client.rb

```diff
@@ -92,6 +92,8 @@ def perform_request_with_retries(method, url, request_args)
       else
         exception_to_reraise = nil
         break
+      ensure
+        set_response(raw_response)
```

Comment:
Since an error is raised (if present) when calling `parse_response_body` and used in how rate limiting is done here, I don't believe we can change that flow without touching other aspects of how the request is handled. 

We could remove the concept of `raw_response` entirely and move setting the response on the initial call, then calling that response thereafter - which simplifies things a bit. At any rate, if parsing the response raises an error we return the error to the client (if not handled by rate limiting) so the last `response` on the client should always be reflective of the response to the request that raised the error. 

```ruby
  def perform_request_with_retries(method, url, request_args)
    parsed_response = nil
    exception_to_reraise = nil

    (@retries + 1).times do |i|
      begin
        set_response(@http_client.send(method, url, request_args))
        parsed_response = parse_response_body(response)
      rescue NationBuilder::RateLimitedError => e
        exception_to_reraise = e
        Kernel.sleep(RETRY_DELAY * 2**i)
      rescue => e
        raise e
      else
        exception_to_reraise = nil
        break
      end
    end

    # If the retry cycle ended with an error, reraise it
    raise exception_to_reraise if exception_to_reraise

    parsed_response
  end
```